custom/src/test/java/com/splunk/opentelemetry/TruncateCommandLineWhenMetricsEnabledTest.java

```diff
@@ -0,0 +1,132 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry;
+
+import static com.splunk.opentelemetry.SplunkConfiguration.METRICS_ENABLED_PROPERTY;
+import static com.splunk.opentelemetry.SplunkConfiguration.METRICS_FULL_COMMAND_LINE;
+import static io.opentelemetry.semconv.resource.attributes.ResourceAttributes.PROCESS_COMMAND_LINE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizer;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class TruncateCommandLineWhenMetricsEnabledTest {
+
+  @Mock ConfigProperties config;
+
+  @Test
+  void shouldNotApplyIfMetricsNotEnabled() {
+    var existing = Resource.create(Attributes.of(PROCESS_COMMAND_LINE, "blargus"));
+
+    when(config.getBoolean(METRICS_ENABLED_PROPERTY, false)).thenReturn(false);
+
+    var testClass = new TruncateCommandLineWhenMetricsEnabled.CommandLineTruncator();
+    var result = testClass.apply(existing, config);
+    assertSame(existing, result);
+  }
+
+  @Test
+  void shouldNotApplyIfNoCommandline() {
+    var existing = Resource.create(Attributes.of(PROCESS_COMMAND_LINE, null));
+    when(config.getBoolean(METRICS_ENABLED_PROPERTY, false)).thenReturn(true);
+
+    var testClass = new TruncateCommandLineWhenMetricsEnabled.CommandLineTruncator();
+
+    var result = testClass.apply(existing, config);
+    assertSame(existing, result);
+  }
+
+  @Test
+  void doesntTruncateWhenCommandlineShort() {
+    var cmd = "c:\\java.exe runme.jar";
+    var existing = Resource.create(Attributes.of(PROCESS_COMMAND_LINE, cmd));
+
+    when(config.getBoolean(METRICS_ENABLED_PROPERTY, false)).thenReturn(true);
+
+    var testClass = new TruncateCommandLineWhenMetricsEnabled.CommandLineTruncator();
+    var result = testClass.apply(existing, config);
+
+    assertSame(existing, result);
+    assertThat(result.getAttribute(PROCESS_COMMAND_LINE)).isEqualTo(cmd);
+  }
+
+  @Test
+  void shouldNotApplyIfConfigItemOverrides() {
+    var existing = Resource.create(Attributes.of(PROCESS_COMMAND_LINE, "blargus"));
```

Comment:
WDYT about creating a resource that contains some other attribute, in addition to the command line -- this would verify that we're not losing any of the other resource attributes, and cover the `existing.merge()` case.

Reply:
Yup, a good idea. Will do.